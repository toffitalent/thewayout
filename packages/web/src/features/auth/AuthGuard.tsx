import { Box, Spinner, useEventCallback } from '@disruptive-labs/ui';
import { useRouter } from 'next/router';
import React, { useEffect, useState } from 'react';
import { UserType } from '@two/shared';
import { useAppAuth, useAppSelector, useAuthRedirect } from '@app/hooks';
import { selectAuthUser } from './reducer';

export interface AuthGuardProps {
  children: React.ReactNode;
  scopes?: string[];
  type: UserType;
}

export function AuthGuard({ children, scopes, type }: AuthGuardProps) {
  const auth = useAppAuth();
  const user = useAppSelector(selectAuthUser);
  const redirect = useAuthRedirect(false);
  const router = useRouter();
  const [isAuthorized, setIsAuthorized] = useState(false);

  const navigateToLogin = useEventCallback(() => {
    router.replace({
      pathname: '/login/',
      query: isAuthorized
        ? undefined
        : {
            return_to: router.asPath,
          },
    });
  });

  useEffect(() => {
    if (auth.isLoading || !router.isReady) return;

    if (!auth.user) {
      navigateToLogin();
    } else if (user && auth.user?.roles?.includes('client') && !user.client) {
      router.push('/client/create-profile');
      setIsAuthorized(true);
    } else if (user && auth.user?.roles?.includes('employer') && !user.employer) {
      router.push('/employer/create-profile');
      setIsAuthorized(true);
    } else if (user && auth.user?.roles?.includes('rsp') && !user.rspAccount?.isProfileFilled) {
      router.push('/ssp/create-profile');
      setIsAuthorized(true);
    } else if (!auth.user?.roles?.includes(type)) {
      redirect();
    } else if (scopes && !auth.user?.scope.split(' ').some?.((r) => scopes.includes(r))) {
      router.replace('/403/');
    } else {
      // Authorized
      setIsAuthorized(true);
    }
    // router is unstable
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [auth.isAuthenticated, auth.isLoading, auth.user, router.isReady, scopes, user]);

  return isAuthorized ? (
    // eslint-disable-next-line react/jsx-no-useless-fragment
    <>{children}</>
  ) : (
    <Box pv={20} textAlign="center">
      <Spinner color="primary" />
    </Box>
  );
}
