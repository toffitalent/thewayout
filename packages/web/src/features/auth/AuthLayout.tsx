import { classNames, SkipNavContent, SkipNavLink } from '@disruptive-labs/ui';
import Image from 'next/legacy/image';
import Link from 'next/link';
import React from 'react';
import logoImg from '@app/assets/images/logo.png';
import styles from './AuthLayout.module.scss';

export interface AuthLayoutProps {
  children?: React.ReactNode;
}

export function AuthLayout({ children }: AuthLayoutProps) {
  return (
    <>
      <SkipNavLink>Skip to content</SkipNavLink>
      <div className={styles.header}>
        <Link href="/" passHref className={styles.logo}>
          <Image src={logoImg} alt="" width={0} height={0} />
        </Link>
      </div>
      <SkipNavContent />
      <main className={classNames(styles.content)}>{children}</main>
    </>
  );
}
