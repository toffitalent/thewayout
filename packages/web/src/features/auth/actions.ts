import { CreateUserRequest } from '@two/shared';
import { API } from '@app/api';
import { createApiThunk } from '@app/utils';

export const fetchUser = createApiThunk('auth/fetchUser', API.user.retrieve);
export const forgotPassword = createApiThunk('auth/forgotPassword', API.user.forgotPassword);

export const login = createApiThunk(
  'auth/login',
  async (credentials: Parameters<typeof API.login>[0], { dispatch }) => {
    await API.login(credentials);
    await dispatch(fetchUser());
  },
);

export const logout = createApiThunk('auth/logout', API.logout);

export const resetPassword = createApiThunk(
  'auth/resetPassword',
  async (
    { username, password, token }: { username: string; password: string; token: string },
    { dispatch },
  ) => {
    await API.user.resetPassword(password, token);
    return dispatch(login({ username, password }));
  },
);

export const signUp = createApiThunk(
  'auth/register',
  async (payload: CreateUserRequest, { dispatch }) => {
    await API.user.create(payload);
    return dispatch(login({ username: payload.email, password: payload.password }));
  },
);

export const updateUser = createApiThunk('auth/updateUser', API.user.update);
