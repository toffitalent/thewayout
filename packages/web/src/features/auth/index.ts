export * from './actions';
export * from './AuthGuard';
export * from './AuthProvider';
export * from './ForgotPasswordPage';
export * from './LoginPage';
export * from './reducer';
export * from './ResetPasswordPage';
export * from './RspSignupPage';
export * from './SignupPage';
