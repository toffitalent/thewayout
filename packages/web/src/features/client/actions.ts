import { ListRspClientRequest } from '@two/shared';
import { API } from '@app/api';
import type { State } from '@app/store';
import { createApiThunk } from '@app/utils';

export const createClient = createApiThunk('client/create', API.clientProfile.create);

export const updateClient = createApiThunk('client/update', API.clientProfile.update);

export const listAllJobs = createApiThunk(
  'client/jobs',
  ({ forceReload }: { forceReload?: boolean }, { getState }) => {
    const { ids } = (getState() as State).clientJobs.jobsList;
    const lastId = [...ids].pop() as string;
    return API.job.list({
      ...(!forceReload && { before: lastId }),
    });
  },
);

export const retrieveJob = createApiThunk(
  'client/job',
  ({ jobId, since }: { jobId: string; since?: Date }) =>
    API.job.retrieve(jobId, {
      ...(since ? { since } : {}),
    }),
);

export const applyJob = createApiThunk(
  'client/jobApply',
  ({ jobId, questions }: { jobId: string; questions?: { [key: string]: string }[] }) =>
    API.job.apply(jobId, {
      ...(questions ? { questions } : {}),
    }),
);

export const listAppliedJobs = createApiThunk(
  'client/applications',
  ({ forceReload, userId }: { forceReload?: boolean; userId: string }, { getState }) => {
    const { ids } = (getState() as State).clientJobs.appliedJobs;
    const lastId = [...ids].pop() as string;
    return API.clientProfile.listJobs(userId, {
      ...(!forceReload && { before: lastId }),
    });
  },
);

export const listSuggestedJobs = createApiThunk('client/suggestedJobs', (clientId: string) =>
  API.clientProfile.listSuggestedJobs(clientId),
);

export const listRsp = createApiThunk(
  'rsp/list',
  ({ userId, ...rest }: ListRspClientRequest & { userId: string }) =>
    API.clientProfile.listRsp({ userId, ...rest }),
);

export const retrieveCaseManager = createApiThunk(
  'client/caseManager',
  API.clientProfile.getCaseManager,
);
