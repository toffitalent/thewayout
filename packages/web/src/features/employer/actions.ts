import { Job, JobApplicationStatus } from '@two/shared';
import { API } from '@app/api';
import type { State } from '@app/store';
import { createApiThunk } from '@app/utils';

export const createEmployer = createApiThunk('employer/create', API.employer.create);

export const updateEmployer = createApiThunk('employer/update', API.employer.update);

export const createJob = createApiThunk('employer/job/create', API.job.create);

export const listJobs = createApiThunk(
  'employer/jobs',
  ({ forceReload, userId }: { forceReload?: boolean; userId: string }, { getState }) => {
    const { ids } = (getState() as State).employerJobs.jobs;
    const lastId = [...ids].pop() as string;
    return API.employer.listJobs(userId, {
      ...(!forceReload && { before: lastId }),
    });
  },
);

export const updateListJobs = createApiThunk(
  'employer/jobs/listUpdated',
  ({ userId }: { userId: string }, { getState }) => {
    const { listUpdatedAt, ids } = (getState() as State).employerJobs.jobs;
    const firstId = ids[0] as string;
    const lastId = [...ids].pop() as string;
    return API.employer.listJobs(userId, {
      since: listUpdatedAt,
      after: firstId,
      before: lastId,
    });
  },
);

export const retrieveJob = createApiThunk(
  'employer/job',
  ({ jobId, since }: { jobId: string; since?: Date }) =>
    API.job.retrieve(jobId, {
      ...(since ? { since } : {}),
    }),
);

export const listApplications = createApiThunk(
  'employer/listApplications',
  ({ employerId, jobId }: { employerId: string; jobId: string }, { getState }) => {
    const { entities } = (getState() as State).employerJobs.jobsApplications;
    const page = entities[jobId]?.page;
    const isListEmpty = !entities[jobId]?.applications.length;

    return API.employer.listApplications(employerId, jobId, {
      page: isListEmpty ? 0 : page! + 1,
    });
  },
);

export const getClient = createApiThunk(
  'employer/getClient',
  ({ employerId, jobId, clientId }: { employerId: string; jobId: string; clientId: string }) =>
    API.employer.getClient(employerId, jobId, clientId, {}),
);

export const updateJob = createApiThunk(
  'employer/updateJob',
  ({ jobId, patch }: { jobId: string; patch: Partial<Job> }) => API.job.update(jobId, patch),
);

export const interviewClient = createApiThunk(
  'employer/interviewClient',
  ({
    employerId,
    jobId,
    applicationId,
  }: {
    employerId: string;
    jobId: string;
    applicationId: string;
  }) =>
    API.employer.changeApplicationStatus(
      employerId,
      jobId,
      applicationId,
      JobApplicationStatus.interview,
    ),
);

export const hireClient = createApiThunk(
  'employer/hireClient',
  ({
    employerId,
    jobId,
    applicationId,
  }: {
    employerId: string;
    jobId: string;
    applicationId: string;
  }) =>
    API.employer.changeApplicationStatus(
      employerId,
      jobId,
      applicationId,
      JobApplicationStatus.hired,
    ),
);

export const rejectClient = createApiThunk(
  'employer/rejectClient',
  ({
    employerId,
    jobId,
    applicationId,
  }: {
    employerId: string;
    jobId: string;
    applicationId: string;
  }) =>
    API.employer.changeApplicationStatus(
      employerId,
      jobId,
      applicationId,
      JobApplicationStatus.rejected,
    ),
);

export const notAFitClient = createApiThunk(
  'employer/notAFitClient',
  ({
    employerId,
    jobId,
    applicationId,
  }: {
    employerId: string;
    jobId: string;
    applicationId: string;
  }) =>
    API.employer.changeApplicationStatus(
      employerId,
      jobId,
      applicationId,
      JobApplicationStatus.notAFit,
    ),
);
