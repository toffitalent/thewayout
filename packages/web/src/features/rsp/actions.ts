import { ClientType, RspClientStatus } from '@two/shared';
import { API } from '@app/api';
import type { State } from '@app/store';
import { createApiThunk } from '@app/utils';

export const createRsp = createApiThunk('rsp/create', API.rsp.create);

export const updateRsp = createApiThunk('rsp/update', API.rsp.update);

export const createMember = createApiThunk('rsp/createMember', API.rsp.createMember);

export const inviteCaseManager = createApiThunk('rsp/inviteCaseManager', API.rsp.inviteCaseManager);

export const removeInvitation = createApiThunk('rsp/removeInvitation', API.rsp.removeInvitation);

export const removeCaseManager = createApiThunk('rsp/removeCaseManager', API.rsp.removeCaseManager);

export const listCaseManagers = createApiThunk(
  'rsp/caseManagers',
  ({ rspId }: { rspId: string }, { getState }) => {
    const { ids } = (getState() as State).rsp.caseManagers;
    const lastId = [...ids].pop() as string;
    return API.rsp.listCaseManagers(rspId, {
      before: lastId,
    });
  },
);

export const listInvitations = createApiThunk(
  'rsp/invitations',
  ({ rspId }: { rspId: string }, { getState }) => {
    const { ids } = (getState() as State).rsp.invitations;
    const lastId = [...ids].pop() as string;
    return API.rsp.listInvitations(rspId, {
      before: lastId,
    });
  },
);

export const listCaseManagersNames = createApiThunk(
  'rsp/listCaseManagersNames',
  API.rsp.listCaseManagersNames,
);

export const retrieveRspAccount = createApiThunk('rsp/retrieveRspAccount', API.rsp.retrieveAccount);

export const updateCaseManager = createApiThunk('rsp/updateCaseManager', API.rsp.updateCaseManager);

export const listClients = createApiThunk(
  'rsp/clients',
  ({ rspId }: { rspId: string }, { getState }) => {
    const { ids } = (getState() as State).rsp.clients;
    const lastId = [...ids].pop() as string;
    return API.rsp.listClients(rspId, {
      before: lastId,
    });
  },
);

export const listStatisticsClients = createApiThunk(
  'rsp/statisticsClients',
  (
    {
      rspId,
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      forceReload,
      ...clientFilters
    }: Partial<ClientType> & { rspId: string; forceReload?: boolean },
    { getState },
  ) => {
    const { ids } = (getState() as State).rsp.statisticsClients;
    const lastId = [...ids].pop() as string;
    return API.rsp.listClients(rspId, {
      ...clientFilters,
      before: lastId,
    });
  },
);

export const retrieveClient = createApiThunk('rsp/retrieveClient', API.rsp.retrieveClient);

export const declineClient = createApiThunk(
  'rsp/declineClient',
  ({ rspId, clientId }: { rspId: string; clientId: string }) =>
    API.rsp.changeClientStatus(rspId, clientId, RspClientStatus.closed),
);

export const acceptClient = createApiThunk(
  'rsp/acceptClient',
  ({ rspId, clientId }: { rspId: string; clientId: string }) =>
    API.rsp.changeClientStatus(rspId, clientId, RspClientStatus.active),
);

export const addClientNotes = createApiThunk('rsp/addClientNotes', API.rsp.addClientNotes);

export const assignCaseManager = createApiThunk(
  'rsp/assignCaseManager',
  ({
    rspId,
    clientId,
    caseManagerId,
  }: {
    rspId: string;
    clientId: string;
    caseManagerId: string;
  }) => API.rsp.assignCaseManager(rspId, clientId, caseManagerId),
);

export const getStatistics = createApiThunk('rsp/statistics', API.rsp.getStatistics);
