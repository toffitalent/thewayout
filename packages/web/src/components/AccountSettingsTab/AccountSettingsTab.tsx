import {
  Avatar,
  Button,
  Controller,
  Field,
  Flex,
  mergeRefs,
  Text,
  Toast,
  useForm,
} from '@disruptive-labs/ui';
import { ChangeEvent, useCallback, useRef } from 'react';
import { MediaType, UpdateUserRequest } from '@two/shared';
import { API } from '@app/api';
import { PhoneNumberInput } from '@app/components/PhoneNumberInput';
import { selectAuthUser, updateUser } from '@app/features/auth';
import { useAppDispatch, useAppSelector, useErrorIndicator } from '@app/hooks';
import { formatPhoneNumber } from '@app/utils';

export const AccountSettingsTab = ({ isPhone }: { isPhone: boolean }) => {
  const user = useAppSelector(selectAuthUser);
  const dispatch = useAppDispatch();
  const showError = useErrorIndicator();
  const ref = useRef<HTMLInputElement | null>(null);

  const {
    control,
    submitForm,
    setValue,
    reset,
    getValues,
    formState: { isValid, isDirty },
  } = useForm({
    onSubmit: async (values) => {
      handleEditUser(values);
      reset(getValues());
    },
    defaultValues: {
      avatar: user?.avatar || '',
      firstName: user?.firstName || '',
      lastName: user?.lastName || '',
      phone: formatPhoneNumber(user?.phone || ''),
      email: user?.email || '',
      file: null as File | null,
    },
  });
  const {
    control: controlPassword,
    submitForm: submitFormPassword,
    reset: resetPassword,
    getValues: getValuesPassword,
    formState: { isValid: isValidPassword, isDirty: isDirtyPassword },
  } = useForm({
    onSubmit: async (values) => {
      handleEditUser(values);
      resetPassword(getValuesPassword());
    },
    defaultValues: {
      password: '',
      newPassword: '',
      confirm: '',
    },
  });

  const handleEditUser = useCallback(
    async (data: UpdateUserRequest & { file?: File | null; confirm?: string }) => {
      if (user) {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const { avatar, phone, file, confirm, ...patch } = data;
        if (file) {
          const { key } = await API.uploads.create(MediaType.Avatar, file);
          (patch as UpdateUserRequest).avatar = key;
        }

        dispatch(
          updateUser({
            ...patch,
            ...(phone && { phone: phone.replace(/[^0-9]/g, '') }),
          }),
        )
          .unwrap()
          .catch((e) => {
            showError(e);
          })
          .then(() => {
            Toast.success('Account Settings updated!!');
          });
      }
    },
    [dispatch, showError, user],
  );

  const handleImgChange = (event: ChangeEvent<HTMLInputElement>) => {
    if (event.target.files) {
      setValue('avatar', URL.createObjectURL(event.target.files[0]), { shouldDirty: true });
      setValue('file', event.target.files[0]);
    }
  };

  return (
    <>
      <Flex mt={8} direction="row" borderBottom pb={5} container>
        <Flex item xs={12} md={4} pr={5} mb={8}>
          <Text fontSize="xl" fontWeight="700">
            Personal Information
          </Text>
          <Text fontSize="sm" color="grey.600">
            Use a permanent address where you can receive mail.
          </Text>
        </Flex>
        <Flex item xs={12} md={8}>
          <form onSubmit={submitForm}>
            <Controller
              name="avatar"
              control={control}
              render={({ field: { ref: controllerRef } }) => (
                <Flex direction="row" mb={10}>
                  <Avatar
                    size="2xl"
                    name={`${user?.firstName} ${user?.lastName}`}
                    src={getValues('avatar') || user?.avatar}
                  />
                  <Flex mt={5} ml={5}>
                    <Button variant="outline" mb={2} onClick={() => ref.current?.click()}>
                      Edit Picture
                    </Button>
                    <input
                      type="file"
                      accept="image/jpg, image/png, image/gif"
                      hidden
                      ref={mergeRefs(ref, controllerRef)}
                      onChange={handleImgChange}
                    />
                    <Text fontSize="sm" color="grey.600">
                      JPG, GIF or PNG. 1MB max.
                    </Text>
                  </Flex>
                </Flex>
              )}
            />
            <Field
              name="firstName"
              type="text"
              label="First Name"
              fluid
              control={control}
              validate={{ required: true }}
              width="1/2"
              display="inline-block"
              pr={1}
            />
            <Field
              name="lastName"
              type="text"
              label="Last Name"
              fluid
              control={control}
              validate={{ required: true }}
              width="1/2"
              display="inline-block"
              pl={1}
            />
            <Field
              name="email"
              type="text"
              label="Email"
              fluid
              control={control}
              validate={{ required: true, email: true }}
              width={isPhone ? '1/2' : 'full'}
              display="inline-block"
              pr={1}
            />
            {isPhone && (
              <PhoneNumberInput
                name="phone"
                control={control}
                label="Phone"
                fluid
                width="1/2"
                display="inline-block"
                pl={1}
                setPhoneValue={(value: string) => {
                  setValue('phone', value, {
                    shouldDirty: true,
                  });
                }}
              />
            )}
            <Button colorScheme="primary" type="submit" mt={8} disabled={!isValid || !isDirty}>
              Save
            </Button>
          </form>
        </Flex>
      </Flex>
      <Flex mt={8} direction="row" container>
        <Flex item xs={12} md={4} pr={5} mb={8}>
          <Text fontSize="xl" fontWeight="700">
            Security
          </Text>
          <Text fontSize="sm" color="grey.600">
            Enter your account password.
          </Text>
        </Flex>
        <Flex item xs={12} md={4}>
          <form onSubmit={submitFormPassword}>
            <Field
              name="password"
              type="password"
              label="Current Password"
              validate={{ required: true }}
              fluid
              control={controlPassword}
            />
            <Field
              name="newPassword"
              type="password"
              label="New Password"
              validate={{ minLength: 8 }}
              fluid
              control={controlPassword}
              width="1/2"
              display="inline-block"
              pr={1}
            />
            <Field
              name="confirm"
              type="password"
              label="Confirm Password"
              validate={{
                minLength: 8,
                validate: (value) =>
                  value !== getValuesPassword('newPassword')
                    ? 'Passwords do not match.'
                    : undefined,
              }}
              fluid
              control={controlPassword}
              width="1/2"
              display="inline-block"
              pl={1}
            />
            <Button
              colorScheme="primary"
              type="submit"
              mt={8}
              disabled={!isValidPassword || !isDirtyPassword}
            >
              Save
            </Button>
          </form>
        </Flex>
      </Flex>
    </>
  );
};
