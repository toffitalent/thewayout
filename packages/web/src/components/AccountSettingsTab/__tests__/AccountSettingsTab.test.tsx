import userEvent from '@testing-library/user-event';
import * as actions from '@app/features/auth/actions';
import { act, fireEvent, fixtures, mockState, mockThunks, render, waitFor } from '@test';
import { AccountSettingsTab } from '../AccountSettingsTab';

const initialState = {
  ...mockState,
  auth: {
    user: { rspAccount: fixtures.rspAccountMember },
  },
};

describe('AccountSettingsTab', () => {
  test('renders correctly', async () => {
    const { container, findByText } = render(<AccountSettingsTab isPhone />, { initialState });

    await waitFor(async () => expect(await findByText('First Name')).toBeInTheDocument());
    expect(container).toMatchSnapshot();
  });

  test('handles submission for personal information', async () => {
    const [updateUser] = mockThunks(actions, ['updateUser']);
    const { getAllByRole, findByText, dispatch } = render(<AccountSettingsTab isPhone />, {
      initialState,
    });

    await userEvent.type(await findByText('First Name'), 'firstname', { delay: 1 });
    await userEvent.type(await findByText('Last Name'), 'lastname', { delay: 1 });
    await userEvent.type(await findByText('Email'), 'email@email.com', { delay: 1 });
    await userEvent.type(await findByText('Phone'), '3344444444', { delay: 1 });

    await act(() => fireEvent.click(getAllByRole('button', { name: 'Save' })[0]));

    expect(updateUser).toBeCalledWith({
      firstName: 'firstname',
      lastName: 'lastname',
      email: 'email@email.com',
      phone: '3344444444',
    });
    expect(dispatch).toBeCalled();
  });

  test('handles submission for security', async () => {
    const [updateUser] = mockThunks(actions, ['updateUser']);
    const { getAllByRole, findByText, dispatch } = render(<AccountSettingsTab isPhone />, {
      initialState,
    });

    await userEvent.type(await findByText('Current Password'), 'password1', { delay: 1 });
    await userEvent.type(await findByText('New Password'), 'password2', { delay: 1 });
    await userEvent.type(await findByText('Confirm Password'), 'password2', { delay: 1 });

    await act(() => fireEvent.click(getAllByRole('button', { name: 'Save' })[1]));

    expect(updateUser).toBeCalledWith({
      password: 'password1',
      newPassword: 'password2',
    });
    expect(dispatch).toBeCalled();
  });

  test('not shows phone input', async () => {
    const { queryByText } = render(<AccountSettingsTab isPhone={false} />, { initialState });
    expect(queryByText('Phone')).not.toBeInTheDocument();
  });
});
