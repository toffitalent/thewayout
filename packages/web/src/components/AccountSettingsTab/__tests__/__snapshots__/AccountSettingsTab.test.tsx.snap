// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSettingsTab renders correctly 1`] = `
<div>
  <div
    class="container borderBottom flexDirection-row marginTop-8 paddingBottom-5"
  >
    <div
      class="item bp-xs-12 bp-md-4 marginBottom-8 paddingRight-5"
    >
      <p
        class="text fontSize-xl fontWeight-700"
      >
        Personal Information
      </p>
      <p
        class="text color-grey-600 fontSize-sm"
      >
        Use a permanent address where you can receive mail.
      </p>
    </div>
    <div
      class="item bp-xs-12 bp-md-8"
    >
      <form>
        <div
          class="flexDirection-row marginBottom-10"
        >
          <div
            class="avatar s-2xl"
          >
            <div
              aria-label="undefined undefined"
              class="initials"
            >
              uu
            </div>
          </div>
          <div
            class="marginTop-5 marginLeft-5"
          >
            <button
              class="button md outline c-primary marginBottom-2"
              type="button"
            >
              Edit Picture
            </button>
            <input
              accept="image/jpg, image/png, image/gif"
              hidden=""
              type="file"
            />
            <p
              class="text color-grey-600 fontSize-sm"
            >
              JPG, GIF or PNG. 1MB max.
            </p>
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingRight-1"
          role="group"
        >
          <label
            class="label"
            for=":r0:"
          >
            First Name
          </label>
          <div
            class="root primary md fluid md"
          >
            <input
              class="input"
              id=":r0:"
              name="firstName"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingLeft-1"
          role="group"
        >
          <label
            class="label"
            for=":r1:"
          >
            Last Name
          </label>
          <div
            class="root primary md fluid md"
          >
            <input
              class="input"
              id=":r1:"
              name="lastName"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingRight-1"
          role="group"
        >
          <label
            class="label"
            for=":r2:"
          >
            Email
          </label>
          <div
            class="root primary md fluid md"
          >
            <input
              class="input"
              id=":r2:"
              name="email"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingLeft-1"
          role="group"
        >
          <label
            class="label"
            for=":r3:"
          >
            Phone
          </label>
          <div
            class="root primary md fluid md"
          >
            <input
              class="input"
              id=":r3:"
              maxlength="14"
              name="phone"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="button md solid c-primary disabled marginTop-8"
          disabled=""
          type="submit"
        >
          Save
        </button>
      </form>
    </div>
  </div>
  <div
    class="container flexDirection-row marginTop-8"
  >
    <div
      class="item bp-xs-12 bp-md-4 marginBottom-8 paddingRight-5"
    >
      <p
        class="text fontSize-xl fontWeight-700"
      >
        Security
      </p>
      <p
        class="text color-grey-600 fontSize-sm"
      >
        Enter your account password.
      </p>
    </div>
    <div
      class="item bp-xs-12 bp-md-4"
    >
      <form>
        <div
          class="control"
          role="group"
        >
          <label
            class="label"
            for=":r4:"
          >
            Current Password
          </label>
          <div
            class="root primary md fluid md"
          >
            <input
              class="input"
              id=":r4:"
              name="password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingRight-1"
          role="group"
        >
          <label
            class="label"
            for=":r5:"
          >
            New Password
          </label>
          <div
            class="root primary md fluid md"
          >
            <input
              class="input"
              id=":r5:"
              name="newPassword"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="control display-inline-block width-1/2 paddingLeft-1"
          role="group"
        >
          <label
            class="label"
            for=":r6:"
          >
            Confirm Password
          </label>
          <div
            class="root primary md fluid md"
          >
            <input
              class="input"
              id=":r6:"
              name="confirm"
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          class="button md solid c-primary disabled marginTop-8"
          disabled=""
          type="submit"
        >
          Save
        </button>
      </form>
    </div>
  </div>
</div>
`;
